Adyen/Helpers/UIViewConstraintsHelper.swift

```diff
@@ -54,5 +97,32 @@ extension AdyenScope where Base: UIView {
         anchor(inside: containerView, with: insets)
         return containerView
     }
+}
 
+/// Edges combined with their inset values.
+/// :nodoc:
+public enum AnchorEdge: Hashable {
+    
+    /// Top edge with its inset value.
+    case top(CGFloat)
+    
+    /// Left edge with its inset value.
+    case left(CGFloat)
+    
+    /// Bottom edge with its inset value.
+    case bottom(CGFloat)
+    
+    /// Right edge with its inset value.
+    case right(CGFloat)
+}
+
+/// An enum to specify an anchor source
+/// :nodoc:
+public enum LayoutAnchorSource {
```

Comment:
can we do something like
```swift
public enum LayoutAnchorSource {
    
    /// Regular `UIView` object
    case view(UIView)
    
    /// Specified layout guide such as Layout Margins or Safe Area.
    case layoutGuide(UILayoutGuide)
    
    var anchored: AnyAnchored {
        switch self {
        case .view(let uIView):
            return uIView
        case .layoutGuide(let uILayoutGuide):
            return uILayoutGuide
        }
    }
}

protocol AnyAnchored {
    var topAnchor: NSLayoutYAxisAnchor { get }
    var leadingAnchor: NSLayoutXAxisAnchor { get }
    var bottomAnchor: NSLayoutYAxisAnchor { get }
    var trailingAnchor: NSLayoutXAxisAnchor { get }
}

extension UILayoutGuide: AnyAnchored {}
extension UIView: AnyAnchored {}

@discardableResult
public func anchor(inside anchorSource: LayoutAnchorSource,
                   onEdges edges: Set<AnchorEdge> = [.top(0),
                                                     .left(0),
                                                     .bottom(0),
                                                     .right(0)]) -> [NSLayoutConstraint] {
    base.translatesAutoresizingMaskIntoConstraints = false
    
    let anchoredSource = anchorSource.anchored
    
    let constraints = edges.map { edge in
        constraint(for: edge, anchoredSource: anchoredSource)
    }
    
    NSLayoutConstraint.activate(constraints)
    return constraints
}

func constraint(for edge: AnchorEdge, anchoredSource: AnyAnchored) -> NSLayoutConstraint {
    switch edge {
    case .top(let padding):
        return base[keyPath: \.topAnchor]
            .constraint(equalTo: anchoredSource[keyPath: \.topAnchor], constant: padding)
    case .left(let padding):
        return base[keyPath: \.leadingAnchor]
            .constraint(equalTo: anchoredSource[keyPath: \.leadingAnchor], constant: padding)
    case .bottom(let padding):
        return base[keyPath: \.bottomAnchor]
            .constraint(equalTo: anchoredSource[keyPath: \.bottomAnchor], constant: padding)
    case .right(let padding):
        return base[keyPath: \.trailingAnchor]
            .constraint(equalTo: anchoredSource[keyPath: \.trailingAnchor], constant: padding)
    }
}
```

Reply:
Nice to reduce the complexity of the function. But couldn't get the benefit of the keypath here. In fact seems harder to read

Reply:
I was thinking more in a sense that the `source` and the `base` both have the same set of anchors, so i wanted to have a single interface for them. we can make extend this further like this, but I don't like that we need 2 functions here
```swift
func constraint(for edge: AnchorEdge, anchoredSource: AnyAnchored) -> NSLayoutConstraint {
    switch edge {
    case let .top(padding):
        return constraint(for: \.topAnchor, source: anchoredSource, padding: padding)
    case let .left(padding):
        return constraint(for: \.leadingAnchor, source: anchoredSource, padding: padding)
    case let .bottom(padding):
        return constraint(for: \.bottomAnchor, source: anchoredSource, padding: padding)
    case let .right(padding):
        return constraint(for: \.trailingAnchor, source: anchoredSource, padding: padding)
    }
}

func constraint(
    for keyPath: KeyPath<AnyAnchored, NSLayoutXAxisAnchor>,
    source: AnyAnchored,
    padding: CGFloat
) -> NSLayoutConstraint {
    base[keyPath: keyPath]
        .constraint(equalTo: source[keyPath: keyPath], constant: padding)
}

func constraint(
    for keyPath: KeyPath<AnyAnchored, NSLayoutYAxisAnchor>,
    source: AnyAnchored,
    padding: CGFloat
) -> NSLayoutConstraint {
    base[keyPath: keyPath]
        .constraint(equalTo: source[keyPath: keyPath], constant: padding)
}
```

Reply:
Overall I think these additions might be too much complexity for the purpose of this function.
Agreed it would reduce function's complexity but adds more IMO.